src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -39,6 +41,36 @@ internal static byte[] SignCore(ReadOnlySpan<byte> content, AsymmetricAlgorithm
             return buffer;
         }
 
+        [UnsupportedOSPlatform("browser")]
+        public static Task<byte[]> SignAsync(Stream detachedContent!!, AsymmetricAlgorithm key!!, HashAlgorithmName hashAlgorithm, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null)
+        {
+            if (!detachedContent.CanRead)
+            {
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotReadable, nameof(detachedContent));
+            }
+
+            if (!detachedContent.CanSeek)
+            {
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotSeekable, nameof(detachedContent));
+            }
```

Comment:
@stephentoub I require that all the Streams provided support Length and Position. Is it wise to throw up-front if the stream is unseekable? 
I think CanSeek doesn't guarantee such thing (nor the opposite) so would be better to just call said properties and let them throw?